### ISSUE TITLE
Texture issues

### ISSUE THREAD

**Issue by cabbant** [ChiselsAndBits/Chisels-and-Bits#974]
Textures are glitched only when using shaders (complementary shaders), I am pretty sure that there arent any other mods causing the problem but I would like to know if there was a fix for this issue

* MC Version: 1.18.2
* C&B Version: ChiselsBits-forge-1.18.2-1.2.111-universal
* Do You have Optifine: Yes
//edit: I have also tried the newest version of C&B and the game crashes after a while running C&B
![2022-07-21_13 51 02](https://user-images.githubusercontent.com/91216303/180208594-04e2cedc-2cee-4be8-a57e-ac6b50db48c0.png)

**Comment by marchermans** [comment]
Do you have quark installed?

Also please make new ticket with the crash thank you

**Comment by cabbant** [comment]
> Do you have quark installed?
> 
> Also please make new ticket with the crash thank you

how can I see the latest crash log? and no, i do not have quark installed
//edit: nvm got it here it i
[crash-2022-07-21_14.07.56-client.txt](https://github.com/ChiselsAndBits/Chisels-and-Bits/files/9160638/crash-2022-07-21_14.07.56-client.txt)
s

**Comment by georgis1979a** [comment]
I have the same glitch problem.Only the version .90 of C&B works!.Any newer version has this glitch(tried with seus ptgi hrr, seus renewed, kappa, complementary shaders ).I cannot update to latest c&b version that fixes errors and issues because of the glitch.Is there anything we can do?

**Comment by RegressiaUK** [comment]
I have this glitch also and we do have quark installed.

**Comment by Axolotlxz** [comment]
I also have this glitch tho I don't have quark. I have also am using shaders and tried 2 different ones vs no shaders of the 1.18.2 forge.
[evidence1.md](https://github.com/ChiselsAndBits/Chisels-and-Bits/files/9275588/evidence1.md)
[evidence2.md](https://github.com/ChiselsAndBits/Chisels-and-Bits/files/9275589/evidence2.md)
[evidence3.md](https://github.com/ChiselsAndBits/Chisels-and-Bits/files/9275590/evidence3.md)
I had some problems with the images working. Sorry.

**Comment by nicholasgrose** [comment]
After observing this in my own game, I decided to look for solutions (or at least workarounds). I discovered that running shaders through Rubidium + Oculus instead of Optifine solved the issue and made shaders work properly with Chisels & Bits. Not sure what it is about Optifine's shader implementation that breaks vertices for the mod, but something isn't working properly.

**Comment by georgis1979a** [comment]
I agree with rubidium and oculus.The only thing with those is they dont support connected textures and Patrix resource pack looks crappy without it.Otherwise I wouldnt touch optifine ever again.They started slacking, ignoring light breaking issues etc.In 1.18.2,only the .90 version of chisels and bits works with optifine .Maybe the settings/actions the author did on that version could be implemented in the newer C&B versions.  Hopefully,If the mod gets updated to 1.19.2,it will working with shaders.I HOPE someone will make a mod for oculus connected textures!!

**Comment by marchermans** [comment]
> I agree with rubidium and oculus.The only thing with those is they dont support connected textures and Patrix resource pack looks crappy without it.Otherwise I wouldnt touch optifine ever again.They started slacking, ignoring light breaking issues etc.In 1.18.2,only the .90 version of chisels and bits works with optifine .Maybe the settings/actions the author did on that version could be implemented in the newer C&B versions. Hopefully,If the mod gets updated to 1.19.2,it will working with shaders.I HOPE someone will make a mod for oculus connected textures!!

The point is beyond the .90 we just use vanilla rendering and face generation.
So we should be directly compatible with optifine.

**Comment by nicholasgrose** [comment]
> I agree with rubidium and oculus.The only thing with those is they dont support connected textures and Patrix resource pack looks crappy without it.Otherwise I wouldnt touch optifine ever again.They started slacking, ignoring light breaking issues etc.In 1.18.2,only the .90 version of chisels and bits works with optifine .Maybe the settings/actions the author did on that version could be implemented in the newer C&B versions.  Hopefully,If the mod gets updated to 1.19.2,it will working with shaders.I HOPE someone will make a mod for oculus connected textures!!

I definitely second the need for a mod for connected textures with Rubidium and Oculus. Fabric/Quilt has Continuity, but there is no equivalent for Forge, as far as I know.

> The point is beyond the .90 we just use vanilla rendering and face generation. So we should be directly compatible with optifine.

It's definitely something that Optifine changes when shaders are enabled. Optifine has no issues without shaders, but even enabling the internal shaders immediately causes chisel blocks to look wrong, despite no obvious visual changes otherwise to the game.
